spec/fixtures/mock_upstream.lua

```diff
@@ -1,12 +1,94 @@
-local utils = require "kong.tools.utils"
+local utils      = require "kong.tools.utils"
+local cjson_safe = require "cjson.safe"
+local cjson      = require "cjson"
+local ws_server  = require "resty.websocket.server"
+
+
+local function string_trim(s)
+  local _,i1 = s:find("^%s*")
+  return s:sub(i1 + 1, s:find("%s*$") - 1)
+end
+
+
+local function string_beginswith(s, prefix)
+  return s:sub(1, #prefix) == prefix
+end
+
+
+local function parse_multipart_form_params(body, content_type)
```

Comment:
Hmmm, any reason for not using lua-multipart (already one of our dependencies) here?